core/src/main/java/jenkins/install/InstallUtil.java

```diff
@@ -74,7 +74,7 @@
      */
     public static InstallState getInstallState() {
         // Support a simple state override. Useful for testing.
-        String stateOverride = System.getenv("jenkins.install.state");
+        String stateOverride = System.getProperty("jenkins.install.state", System.getenv("jenkins.install.state"));
```

Comment:
Windows development!  because this will not work on [Linux with Bash](http://unix.stackexchange.com/questions/93532/exporting-a-variable-with-dot-in-it#answer-93533) as you can not have environment variables with dots in them :-)